domehunter/dome_control.py

```diff
@@ -0,0 +1,674 @@
+
+        # if updated delta_az is positive, direction is CW
+        if delta_az > 0:
+            # converted delta_az to equivilant in encoder ticks
+            ticks = self._az_to_ticks(delta_az)
+            target_position = self.encoder_count + ticks
+            self._move_cw()
+            # wait until encoder count matches desired delta az
+            while self.encoder_count < target_position:
+                if self.testing:
+                    # if testing simulate a tick for every cycle of while loop
+                    self._simulate_ticks(num_ticks=1)
+            self._stop_moving()
+            # compare original count to current just in case we got more ticks
+            # than we asked for
+            old_encoder_count = target_position - ticks
+            # update dome_az based on the actual number of ticks counted
+            self.dome_az += self._ticks_to_az(
+                self.encoder_count - old_encoder_count)
+            # take mod360 of dome_az to keep 0 <= dome_az < 360
+            self.dome_az %= 360
+            # update encoder_count to match the dome_az
+            self.encoder_count = self._az_to_ticks(self.dome_az)
+
+        # if updated delta_az is negative, direction is CCW
+        if delta_az < 0:
+            # converted delta_az to equivilant in encoder ticks
+            ticks = self._az_to_ticks(delta_az)
+            target_position = self.encoder_count + ticks
+            self._move_ccw()
+            # wait until encoder count matches desired delta az
+            while self.encoder_count >= target_position:
+                if self.testing:
+                    # if testing simulate a tick for every cycle of while loop
+                    self._simulate_ticks(num_ticks=1)
+                else:
+                    # micro break to spare the little rpi cpu
+                    time.sleep(0.1)
+            self._stop_moving()
+            # compare original count to current, just in case we got more ticks
+            # than we asked for
+            old_encoder_count = target_position - ticks
+            # update dome_az based on the actual number of ticks counted
+            self.dome_az += self._ticks_to_az(
+                self.encoder_count - old_encoder_count)
+            # take mod360 of dome_az to keep 0 <= dome_az < 360
+            self.dome_az %= 360
+            # update encoder_count to match the dome_az
+            self.encoder_count = self._az_to_ticks(self.dome_az)
+
+    def calibrate_dome_encoder_counts(self, num_cal_rotations=2):
+        """
+        Calibrate the encoder (determine degrees per tick).
+
+        Parameters
+        ----------
+        num_cal_rotations : integer
+            Number of rotations to perform to calibrate encoder.
+
+        """
+        # rotate the dome until we hit home, to give reference point
+        self.find_home()
+
+        # now set dome to rotate n times so we can determine the number of
```

Comment:
This comment should probably go right above the `while` loop.
```suggestion
        # now set dome to rotate num_cal_rotation times so we can determine the number of
```